src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,133 @@
+    private static final String AT_LEAST_ONE_ROW_EXCEPTION
+            = "Expected to insert at least one row, but inserted 0 rows";
+    private static final String ADD_BOOK_EXCEPTION = "Cannot add new book: ";
+    private static final String GET_BOOK_EXCEPTION = "Cannot get a book by id: ";
+    private static final String GET_ALL_BOOKS_EXCEPTION = "Cannot get books";
+    private static final String UPDATE_BOOK_EXCEPTION = "Cannot update book by id ";
+    private static final String DELETE_BOOK_EXCEPTION = "Cannot delete by id ";
+    private static final String NO_BOOK_EXCEPTION = "There are no book by id: ";
+    private static final String NULL_FIELDS_EXCEPTION = "Title name or price cannot be null";
+    private static final String NEGATIVE_ID_EXCEPTION = "ID cannot be negative";
+    private static final String NO_RECORDS_EXCEPTION = "There are no records in DB";
+    private static final int LESS_THAN_ONE = 1;
+    private static final int AT_LEAST_ONE = 0;
+    private static final int LESS_THAN_ZERO = 0;
+    private static final int ZERO = 0;
+
+    @Override
+    public Book create(Book book) {
+        if (book.getTitle() == null || book.getPrice() == null) {
+            throw new RuntimeException(NULL_FIELDS_EXCEPTION);
+        }
+        String sqlUpdate = "INSERT INTO books (title, price) VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+             PreparedStatement statement = connection.prepareStatement(sqlUpdate, Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+            int affectedRows = statement.executeUpdate();
+            if (affectedRows < LESS_THAN_ONE) {
+                throw new RuntimeException(AT_LEAST_ONE_ROW_EXCEPTION);
+            }
+            ResultSet generatedKeys = statement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(1, Long.class);
+                book.setId(id);
+            }
+            return book;
+        } catch (SQLException e) {
+            throw new DataProcessingException(ADD_BOOK_EXCEPTION + book, e);
+        }
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        if (id < LESS_THAN_ZERO) {
+            throw new RuntimeException(NEGATIVE_ID_EXCEPTION);
+        }
+        String sqlQuery = "SELECT * FROM books WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+             PreparedStatement statement = connection.prepareStatement(sqlQuery)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                String model = resultSet.getString("title");
+                BigDecimal price = resultSet.getObject("price", BigDecimal.class);
+                return Optional.of(new Book(id, model, price));
```

Comment:
You can take out the book variable before the try block, then use book = new Book(your parsed fields) in the try block, and return optional.ofNullable(book) after the catch block